tests/runtests/skip-sosreport-dir/runtest.sh

```diff
@@ -0,0 +1,135 @@
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+. /usr/share/beakerlib/beakerlib.sh
+. ../aux/lib.sh
+
+TEST="skip-sosreport-test"
+PACKAGE="abrt"
+SOSREPORT_EVENT="/etc/libreport/events.d/sosreport_event.conf"
+
+rlJournalStart
+    rlPhaseStartSetup
+        check_prior_crashes
+        TmpDir=$(mktemp -d)
+        pushd "$TmpDir" || exit
+
+        # generate large file to fill problem directories space
+        dd if=/dev/urandom of=hugh_mongous count=13337 bs=1024
+
+        if [[ -f "$SOSREPORT_EVENT" ]]; then
+            rlFileBackup --namespace sos "$SOSREPORT_EVENT"
+            rm -f "$SOSREPORT_EVENT"
+        fi
+    rlPhaseEnd
+
+    rlPhaseStartTest "[1/3] delete dir"
+        prepare
+        generate_crash
+        wait_for_hooks
+        get_crash_path
+
+        rlRun "cp $TmpDir/hugh_mongous $crash_PATH" 0 "Fill problem directory with a huge file"
+
+        rlRun "abrt-action-trim-files -vvv -d 1M:$ABRT_CONF_DUMP_LOCATION &> $TmpDir/trim-files-output" 0
+        rlRun "cat $TmpDir/trim-files-output"
+
+        rlAssertNotExists "$crash_PATH"
+        rlAssertGrep "deleting '$(basename $crash_PATH)'" "$TmpDir/trim-files-output"
+
+        rlRun "abrt-cli rm $crash_PATH" 1 "Remove crash directory"
+    rlPhaseEnd
+
+    rlPhaseStartTest "[2/3] skip dir with sosreport, delete it if without sosreport"
+        prepare
+        generate_crash
+        wait_for_hooks
+        get_crash_path
+
+        rlRun "cp $TmpDir/hugh_mongous $crash_PATH" 0 "Fill problem directory with a huge file"
+        rlRun "touch $crash_PATH/sosreport.log" 0 "Create FAKE sosreport.log file"
+
+        rlRun "abrt-action-trim-files -vvv -d 1M:$ABRT_CONF_DUMP_LOCATION &> $TmpDir/trim-files-skip-output" 0
+        rlRun "cat $TmpDir/trim-files-skip-output"
+
+        rlAssertExists "$crash_PATH"
+        rlAssertGrep "$crash_PATH': sosreport is being generated." "$TmpDir/trim-files-skip-output"
+
+        rlRun "rm $crash_PATH/sosreport.log" 0 "Remove FAKE sosreport.log file"
+
+        rlRun "abrt-action-trim-files -vvv -d 1M:$ABRT_CONF_DUMP_LOCATION &> $TmpDir/trim-files-delete-output" 0
+        rlRun "cat $TmpDir/trim-files-delete-output"
+
+        rlAssertNotExists "$crash_PATH"
+        rlAssertGrep "deleting '$(basename $crash_PATH)'" "$TmpDir/trim-files-delete-output"
+
+        rlRun "abrt-cli rm $crash_PATH" 1 "Remove crash directory"
```

Comment:
Can you use `remove_problem_directory` here?

Reply:
I painstakingly replaced this everywhere should we ever change the CLI again.

Reply:
Of course, in cases where you’re removing multiple directories, this won’t work, but oh well.

Reply:
> Can you use `remove_problem_directory` here?

I am testing for a failure (1). 

I could maybe modify the function to accept different $crash_PATHs and return codes.

Reply:
> > Can you use `remove_problem_directory` here?
> 
> I am testing for a failure (1).

I should change the comment to make it more clear.

Reply:
Oops, pardon my reading comprehension.

Reply:
I think it’s fine how it is, no need for anything else.